### PR TITLE
Show buffer memory usage in Admin Console

### PR DESCRIPTION
**Pull request by labkey-jeckels**
#### Rationale
The JVM may be using a lot of buffer memory, causing instances to crash when the OS runs low on memory. Having some visibility will be helpful while we get full perf monitoring in place

#### Changes
* Avoid infinite mothership reporting on Throwables
* Show buffer pool usages
* Better formatting on memory info page
* Show CPU info and total OS memory

### CODE REVIEW

File: core/src/org/labkey/core/admin/AdminController.java [hunk 1]
```diff
@@ -3406,13 +3410,19 @@ private MemBean(HttpServletRequest request, Set<Object> objectsToIgnore)
...
+            }
+
             // class loader:
             ClassLoadingMXBean classbean = ManagementFactory.getClassLoadingMXBean();
             if (classbean != null)
             {
-                systemProperties.add(new Pair<>("Loaded Class Count", classbean.getLoadedClassCount()));
-                systemProperties.add(new Pair<>("Unloaded Class Count", classbean.getUnloadedClassCount()));
-                systemProperties.add(new Pair<>("Total Loaded Class Count", classbean.getTotalLoadedClassCount()));
+                systemProperties.add(new Pair<>("Loaded Class Count", Formats.commaf0.format(classbean.getLoadedClassCount())));
```
Comment (labkey-matthewb) [inline core/src/org/labkey/core/admin/AdminController.java]: Yikes.  Formats are not thread-safe.  We should kill those constants (use getters or methods with synchronize?)

**Review by labkey-matthewb — APPROVED** [review]
Nice.  Some of the charts have weird scales (/0 or /-1?).  But otherwise looks good.

File: core/src/org/labkey/core/admin/AdminController.java (hunk 1 above)
Comment (labkey-jeckels) [reply core/src/org/labkey/core/admin/AdminController.java]: I did some digging on this earlier. The general consensus seems to be that if you're not mutating the formatter itself, they're safe to share across threads. But I'm not sure I'm fully convinced of that argument. We could also use ThreadLocals.

Comment (labkey-matthewb) [reply core/src/org/labkey/core/admin/AdminController.java]: I'm not sure what the rules are but I fixed two teamcity failures recently related to this.  I'm not sure if this has become problematic with the new JVM?

**Comment by labkey-jeckels** [comment]
I made the plots without real data to show look better. Not awesome, but good enough for this page.

File: core/src/org/labkey/core/admin/AdminController.java (hunk 1 above)
Comment (labkey-jeckels) [reply core/src/org/labkey/core/admin/AdminController.java]: Interesting. I purged references to the shared Formats from this class but am not planning to eliminate all usages at this moment.

Comment (labkey-adam) [reply core/src/org/labkey/core/admin/AdminController.java]: We've seen problems with shared formats and `parse()`, but never with `format()`.

Comment (labkey-matthewb) [reply core/src/org/labkey/core/admin/AdminController.java]: Right.  Good point.

**Pull request merged**